src/backend/distributed/planner/multi_explain.c

```diff
@@ -293,10 +296,60 @@ RemoteExplain(Task *task, ExplainState *es)
 		ShardPlacement *taskPlacement = list_nth(taskPlacementList, placementIndex);
 		char *nodeName = taskPlacement->nodeName;
 		uint32 nodePort = taskPlacement->nodePort;
+		MultiConnection *connection = NULL;
+		PGresult *queryResult = NULL;
+		int connectionFlags = SESSION_LIFESPAN;
+		int executeResult = 0;
 
 		remotePlan->placementIndex = placementIndex;
-		remotePlan->explainOutputList = ExecuteRemoteQuery(nodeName, nodePort,
-														   NULL, explainQuery);
+
+		connection = GetNodeConnection(connectionFlags, nodeName, nodePort);
+
+		/*
+		 * Use a coordinated transaction to skip sending BEGIN in an already
+		 * existing transaction.
+		 */
+		BeginOrContinueCoordinatedTransaction();
+		RemoteTransactionBeginIfNecessary(connection);
+
+		/* start a savepoint for explain */
+		executeResult = ExecuteOptionalRemoteCommand(connection,
```

Comment:
On the other hand, the code would be a bit cleaner here if we used ExecuteCriticalRemoteCommand. If we add `if (connection->pgConn) continue;` after `GetPlacementConnection` then we still get failover if the worker is down.